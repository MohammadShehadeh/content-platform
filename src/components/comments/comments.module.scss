.comment {
  padding: var(--space-md);
}
