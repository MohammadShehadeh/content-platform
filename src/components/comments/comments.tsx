import { Description } from '@/components/description';
import { Card } from '@/components/shared/card';
import { Title } from '@/components/title';
import type { Comment } from '@/services/comments';

import styles from './comments.module.scss';

type CommentsProps = Comment;

export const Comments = ({ name, email, body }: CommentsProps) => {
  return (
    <Card className={styles.comment}>
      <Title asChild>
        <h3>{name}</h3>
      </Title>
      <Description color="muted">{email}</Description>
      <Description size="lg">{body}</Description>
    </Card>
  );
};
