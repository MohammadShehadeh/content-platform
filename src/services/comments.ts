import { COMMENT_API_ROUTE } from '@/constants/apis';
import { safeFetch } from '@/lib/fetch';

import type { Pagination } from './types';

export type Comment = {
  postId: number;
  id: number;
  name: string;
  email: string;
  body: string;
};

/**
 * Get Comments by Post ID
 * @param postId - The ID of the post to get comments for
 * @param offset - The offset of the comments to get
 * @param limit - The limit of the comments to get
 * @returns The comments for the post
 */
export const getCommentsByPostId = async (postId: number, { offset, limit }: Pagination) => {
  const { data, error } = await safeFetch<Comment[]>(
    `${COMMENT_API_ROUTE}?postId=${postId}&_start=${offset}&_limit=${limit}`
  );

  return { data, error };
};
