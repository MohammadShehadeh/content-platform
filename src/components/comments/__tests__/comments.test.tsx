import { render, screen } from '@testing-library/react';

import { Comments } from '@/components/comments';

describe('Comments', () => {
  it('should render with all props correctly', () => {
    render(<Comments name="John Doe" email="john.doe@example.com" body="This is a comment" />);

    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('john.doe@example.com')).toBeInTheDocument();
    expect(screen.getByText('This is a comment')).toBeInTheDocument();
  });

  it('should render with different content', () => {
    render(
      <Comments name="Jane Smith" email="jane.smith@example.com" body="Another comment here" />
    );

    expect(screen.getByText('Jane Smith')).toBeInTheDocument();
    expect(screen.getByText('jane.smith@example.com')).toBeInTheDocument();
    expect(screen.getByText('Another comment here')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(
      <Comments name="John Doe" email="john.doe@example.com" body="This is a comment" />
    );

    expect(container).toMatchSnapshot();
  });
});
