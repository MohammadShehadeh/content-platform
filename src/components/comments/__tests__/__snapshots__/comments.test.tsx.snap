// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments matches snapshot 1`] = `
<div>
  <div
    class="card comment"
  >
    <h3
      class="title lg"
    >
      John Doe
    </h3>
    <p
      class="description md muted"
    >
      john.doe@example.com
    </p>
    <p
      class="description lg default"
    >
      This is a comment
    </p>
  </div>
</div>
`;

exports[`Comments should match snapshot 1`] = `
<div>
  <div
    class="card comment"
  >
    <h3
      class="title lg"
    >
      John Doe
    </h3>
    <p
      class="description md muted"
    >
      john.doe@example.com
    </p>
    <p
      class="description lg default"
    >
      This is a comment
    </p>
  </div>
</div>
`;
